plugin/advisor/pg/advisor_naming_foreign_key_convention.go

```diff
@@ -0,0 +1,175 @@
+	if err != nil {
+		return nil, err
+	}
+
+	format, templateList, maxLength, err := advisor.UnmarshalNamingRulePayloadAsTemplate(ctx.Rule.Type, ctx.Rule.Payload)
+	if err != nil {
+		return nil, err
+	}
+
+	checker := &namingFKConventionChecker{
+		level:        level,
+		title:        string(ctx.Rule.Type),
+		format:       format,
+		maxLength:    maxLength,
+		templateList: templateList,
+	}
+
+	for _, stmtNode := range root {
+		ast.Walk(checker, stmtNode)
+	}
+
+	if len(checker.adviceList) == 0 {
+		checker.adviceList = append(checker.adviceList, advisor.Advice{
+			Status:  advisor.Success,
+			Code:    advisor.Ok,
+			Title:   "OK",
+			Content: "",
+		})
+	}
+
+	return checker.adviceList, nil
+}
+
+type namingFKConventionChecker struct {
+	adviceList   []advisor.Advice
+	level        advisor.Status
+	title        string
+	format       string
+	maxLength    int
+	templateList []string
+}
+
+type indexMetaData struct {
+	indexName string
+	tableName string
+	metaData  map[string]string
+}
+
+// Visit implements ast.Visitor interface
+func (checker *namingFKConventionChecker) Visit(in ast.Node) ast.Visitor {
+	indexDataList := checker.getMetaDataList(in)
+
+	for _, indexData := range indexDataList {
+		regex, err := getTemplateRegexp(checker.format, checker.templateList, indexData.metaData)
+		if err != nil {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.Internal,
+				Title:   "Internal error for foreign key naming convention rule",
+				Content: fmt.Sprintf("%q meet internal error %q", in.Text(), err.Error()),
+			})
+			continue
+		}
+		if !regex.MatchString(indexData.indexName) {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingFKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf("Foreign key in table `%s` mismatches the naming convention, expect %q but found `%s`", indexData.tableName, regex, indexData.indexName),
+			})
+		}
+		if checker.maxLength > 0 && len(indexData.indexName) > checker.maxLength {
+			checker.adviceList = append(checker.adviceList, advisor.Advice{
+				Status:  checker.level,
+				Code:    advisor.NamingFKConventionMismatch,
+				Title:   checker.title,
+				Content: fmt.Sprintf("Foreign key `%s` in table `%s` mismatches the naming convention, its length should within %d characters", indexData.indexName, indexData.tableName, checker.maxLength),
+			})
+		}
+	}
+
+	return checker
+}
+
+// getMetaDataList returns the list of foreign key with metadata.
+func (checker *namingFKConventionChecker) getMetaDataList(in ast.Node) []*indexMetaData {
+	var res []*indexMetaData
+	switch node := in.(type) {
+	case *ast.CreateTableStmt:
+		for _, constraint := range node.ConstraintList {
+			metadata := getForeignKeyMetadata(constraint, node.Name.Name)
+			if metadata != nil {
+				res = append(res, metadata)
+			}
+		}
+	case *ast.AddConstraintStmt:
```

Comment:
We should consider the ALTER TABLE ADD COLUMN statements here. It can create a column constraint. Details at https://www.postgresql.org/docs/14/sql-altertable.html